peertube/docker-compose.yaml

```diff
@@ -0,0 +1,77 @@
+services:
+  peertube:
+    # If you don't want to use the official image and build one from sources:
+    # build:
+    #   context: .
+    #   dockerfile: ./support/docker/production/Dockerfile.bookworm
+    image: chocobozzz/peertube:${PEERTUBE_IMAGE}
+    environment:
+      - PEERTUBE_DB_USERNAME=${PEERTUBE_POSTGRES_USER}
+      - PEERTUBE_DB_PASSWORD=${PEERTUBE_POSTGRES_PASSWORD}
+      - PEERTUBE_DB_SSL=${PEERTUBE_DB_SSL}
+      - PEERTUBE_DB_HOSTNAME=postgres
+      - PEERTUBE_WEBSERVER_HOSTNAME=${PEERTUBE_TRAEFIK_HOST}
+      - PEERTUBE_TRUST_PROXY=${PEERTUBE_TRUST_PROXY}
+      - PEERTUBE_SECRET=${PEERTUBE_SECRET}
+      - PEERTUBE_SMTP_USERNAME=${PEERTUBE_SMTP_USERNAME}
+      - PEERTUBE_SMTP_PASSWORD=${PEERTUBE_SMTP_PASSWORD}
+      - PEERTUBE_SMTP_HOSTNAME=${PEERTUBE_SMTP_HOSTNAME}
+      - PEERTUBE_SMTP_PORT=${PEERTUBE_SMTP_PORT}
+      - PEERTUBE_SMTP_FROM=${PEERTUBE_SMTP_FROM}
+      - PEERTUBE_SMTP_TLS=${PEERTUBE_SMTP_TLS}
+      - PEERTUBE_SMTP_DISABLE_STARTTLS=${PEERTUBE_SMTP_DISABLE_STARTTLS}
+      - PEERTUBE_ADMIN_EMAIL=${PEERTUBE_ADMIN_EMAIL}
+      - POSTFIX_myhostname=${PEERTUBE_POSTFIX_myhostname}
+      - OPENDKIM_DOMAINS=${PEERTUBE_OPENDKIM_DOMAINS}
+      - OPENDKIM_RequireSafeKeys=${PEERTUBE_OPENDKIM_RequireSafeKeys}
+      - PEERTUBE_OBJECT_STORAGE_UPLOAD_ACL_PUBLIC="public-read"
+      - PEERTUBE_OBJECT_STORAGE_UPLOAD_ACL_PRIVATE="private"
+      - PEERTUBE_LOG_LEVEL=${PEERTUBE_LOG_LEVEL}
+    ports:
+      # Comment next line if you don't want to use the live feature
+      - "1935:1935" 
```

Comment:
We don't want to expose any ports, we should create traefik entrypoints if we need additional ports.

Reply:
I'm going to need help with this one.

Reply:
I added the entrypoints in #3f4e49a 
You can use the rtmp or rtmps entrypoints.
Example in forgejo where it uses the ssh entrypoint https://github.com/EnigmaCurry/d.rymcg.tech/blob/033bf8e3331b8ca8dcb945380b4365780f855a7a/forgejo/docker-compose.instance.yaml#L33-L37

Reply:
I didn't add an entrypoint for the 9000 port, because the comment indicated it wasnt for prod use.

Reply:
README should indicate that to enable RTMP you have to enable the associated traefik entrypoint

Reply:
For RTMPS you would have traefik terminate the TLS connection with a lets encrypt cert on port 1936 and forward it unencrypted to the peertube service port 1935.

You only need to set a blank tls section and do the port translation:

```
- "traefik.tcp.services.(@= router @)-rtmps.tls=true" 
- "traefik.tcp.services.(@= router @)-rtmps.loadbalancer.server.port=1935"
```

https://doc.traefik.io/traefik/routing/routers/#general_2

Reply:
I don't understand this. :(

[25b8edd](https://github.com/EnigmaCurry/d.rymcg.tech/pull/260/commits/25b8edde28b3b8e13d08b55f16300e85b035932e) is my attempt at using other code as a template to guess what should be done here.

Reply:
yea that looks right.

Reply:
well youd need two routers to handle both rtmp or rtmps, but it seems like any rtmp client is capable of configuring a port, so I think my commit earlier where I made both entrypoints should be consolidated to a single entrypoint `rtmp` and the user can configure if they want TLS or not themselves.

so for rtmps I think it would be:

```
      #! RTMPS
      #@ if livestreaming == "RTMPS":
      - "traefik.tcp.services.(@= router @)-rtmp.tls=true" 
      #@ end
```

Reply:
I deleted my comment about needing another variable (I know you read email notifications though)

Reply:
yea

Reply:
How does [9d55891](https://github.com/EnigmaCurry/d.rymcg.tech/pull/260/commits/9d5589148474bd799bc310e997392fe7bff66473) look? Peertube works, but I wasn't able to successfully test livestreaming - mainly because I've never livestreamed and don't know how. I started a livestream in Peertube and entered the URL (e.g., rtmp://peertube.thewooskeys.com:1935/live) into VLC, but VLC can't connect. I tried rtp://, http://, and https://. I do have an entrypoint and layer 4 route on the sentry (error free in Traefik dashboard) and an entrypoint on the app server (also error free in Traefik dashboard).

Reply:
looking good. I would test with OBS.

Reply:
Same result: OBS can't connect. So either it doesn't work as it is or, more likely, I don't have it set up correctly.

Reply:
`rtmpdump` also can't connect. I think I must have the route configured incorrectly but I don't yet see how.

Reply:
My bad. I neglected to open 1935 on the sentry's firewall. Everything works! I think this PR can be merged, @EnigmaCurry.

Reply:
nice, you tested with RTMP and RTMPS? OBS accepts the certificate?